### PR TITLE
feat(autoware_lidar_centerpoint): added a check to notify if we are dropping pillars

### PR DESCRIPTION
**Pull request by knzo25**
## Description
If the input pointcloud generates more pillars than the model that handle, we are currently clipping them.
If that happens, we should notify users so they can consider increasing the model parameters.

## Related links

**Parent Issue:**

- Link

## How was this PR tested?
Artificially decreased the maximum amount of voxels/pillars in the config file and made sure that the error message was displayed in a throttled manner.

## Notes for reviewers

If you do not re-generate the engine you will have runtime errors

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: perception/autoware_lidar_centerpoint/lib/centerpoint_trt.cpp [hunk 1]
```diff
@@ -144,6 +144,19 @@ bool CenterPointTRT::detect(
 
   postProcess(det_boxes3d);
 
+  // Check the actual number of pillars after inference to avoid unnecessary synchronization.
+  unsigned int num_pillars = 0;
+  CHECK_CUDA_ERROR(
+    cudaMemcpy(&num_pillars, num_voxels_d_.get(), sizeof(unsigned int), cudaMemcpyDeviceToHost));
+
+  if (num_pillars >= config_.max_voxel_size_) {
+    RCLCPP_WARN(
```
[the remark this answers is not in the document]
Comment (knzo25) [reply perception/autoware_lidar_centerpoint/lib/centerpoint_trt.cpp]: @kminoda 
I did not want to include a node reference in the tensor trt node (nvcc related), but it seems this works fine as well

**Comment by kminoda** [comment]
@knzo25 Would you also update the "How was this PR tested?" section in the description?

**Comment by knzo25** [comment]
@kminoda 
Done

**Review by kminoda — APPROVED** [review]
LGTM, thanks for the quick fix

**Pull request merged**